sudoku.rb

```diff
@@ -0,0 +1,165 @@
+# Sudoku Rules:
+#   Each cell should contain a digit 1-9 such that:
+#     * horizontal row contains each digit exactly once
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [1 2 3  4 5 6  7 8 9]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#     * vertical column contains each digit exactly once
+#       [_ _ _  _ 1 _  _ _ _]
+#       [_ _ _  _ 2 _  _ _ _]
+#       [_ _ _  _ 3 _  _ _ _]
+#
+#       [_ _ _  _ 4 _  _ _ _]
+#       [_ _ _  _ 5 _  _ _ _]
+#       [_ _ _  _ 6 _  _ _ _]
+#
+#       [_ _ _  _ 7 _  _ _ _]
+#       [_ _ _  _ 8 _  _ _ _]
+#       [_ _ _  _ 9 _  _ _ _]
+#     * each 3x3 box contains each digit exactly once
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  1 2 3  _ _ _]
+#       [_ _ _  4 5 6  _ _ _]
+#       [_ _ _  7 8 9  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+
+
+class Sudoku
+
+  #
+  # @param board [Array]
+  #
+  def initialize(board)
+    @board = board
+  end
+
+  #
+  # @return [Boolean] true if board is valid
+  #
+  def solved?
+    #
+    # TODO
+    #
+
+    board = @board
+    row_count = 0
+    valid_count = 0
+    valid = false
+
+    # check if vals in rows are unique
+    uniq_check = Proc.new do |b|
+      b.each do |row|
```

Comment:
instead of passing in the collection of arrays to check, another option is to declare a method that checks a single array. that would enable us to break out of the method if any of the rows failed to meet the check. would potentially save a large number of operations from being run